### DISCUSSION TITLE
module 'tvm.arith._ffi_api' has no attribute 'CreateAnalyzer'

### DISCUSSION THREAD

**Discussion by cuiHL** [mlc-ai/mlc-zh#13] [Q&A]
在命令行执行 `pip install mlc-ai-nightly -f https://mlc.ai/wheels` 后（Successfully installed ），在导包`from tvm.script import tir as T`的过程中报错module 'tvm.arith._ffi_api' has no attribute 'CreateAnalyzer'，请教一下这是什么原因造成的啊，该怎么解决？

**Comment by Hzfengsy** [comment]
我这边没有发现问题，没有麻烦告知一下系统版本、Python版本。

**Reply by cuiHL** [reply]
系统版本Windows 专业版21H2（操作系统内部版本号为19044.1645），python版本3.9.13

**Reply by Hzfengsy** [reply]
感谢你的反馈，并对造成的麻烦表示抱歉。我们目前缺少基于Windows平台的测试，我们会尽快在Windows环境上进行测试和修复。在此之前如果可能，可以转而使用Linux或WSL运行

**Reply by cuiHL** [reply]
好的，谢谢

**Reply by llx-08** [reply]
使用python 3.7.13对应的TVM没有这个问题，使用python 3.10对应的TVM出现了这个问题

**Reply by Hzfengsy** [reply]
Windows请使用Python 3.7和3.8

**Reply by baifanmvp123** [reply]
3.10现在依然报错

**Comment by junrushao** [comment]
这个问题看起来是 libtvm 里面的 global registry 正确注册，不知道可不可以检查一下呢？@Hzfengsy

**Comment by junrushao** [comment]
@cuiHL 请问可以打印一下以下的输出吗？

```
import tvm
print(tvm._ffi.registry.list_global_func_names())
```

**Reply by cuiHL** [reply]
您好，下述为print的输出结果
['runtime.module.loadbinary_VMExecutable', 'runtime.module.loadbinary_metadata', 'runtime.profiling.Ratio', '__tvm_set_device', 'runtime.RuntimeEnabled', 'runtime.TVMSetStream', 'device_api.cpu', 'runtime.GetADTTag', 'runtime.GetDeviceAttr', 'runtime.Array', 'runtime.module.loadbinary_const_loader', 'runtime.ArrayGetItem', 'runtime.ArraySize', 'runtime.profiling.Report', 'profiling.start_timer', 'runtime.GetADTSize', 'runtime.GetADTFields', 'runtime.GetShapeTupleElem', 'runtime.Tuple', 'tvm.rpc.server.remove','runtime.ObjectPtrHash', 'runtime.ADT', 'runtime.module.loadfile_relax.Executable', 'runtime.String', 'runtime.GetFFIString', 'runtime.Map', 'runtime.MapSize', 'runtime.ModuleImportsSize', 'runtime.MapGetItem', 'runtime.MapCount', 'runtime.ModuleImplementsFunction', 'runtime.MapItems', 'profiling.timer.cpu', 'runtime.ShapeTuple', 'runtime.GetShapeTupleSize', 'runtime.SaveParams', 'runtime.ModuleSaveToFile', 'runtime.LoadParams', 'runtime.module.loadfile_so', 'runtime.module.loadbinary_metadata_module', 'runtime.module.loadfile_VMExecutable', 'runtime.module.loadbinary_relax.Executable', 'runtime.profiling.ProfileFunction', 'tvm.contrib.random.uniform', 'runtime.ModuleGetSource','runtime.ModuleGetImport', 'runtime.profiling.DeviceWrapper', 'runtime.ModuleGetTypeKey', 'runtime.ModuleGetFormat', 'runtime.ModuleLoadFromFile', 'runtime.profiling.FromJSON', 'runtime.ModuleIsDSOExportable', 'runtime.TVMArrayAllocWithScope', 'runtime.TVMArrayCreateView', 'runtime.profiling.Duration', 'runtime.DumpTypeTable', 'runtime.profiling.AsTable', 'runtime.profiling.AsCSV', 'tvm.contrib.sort.topk', 'runtime.profiling.AsJSON', 'runtime.profiling.Count', 'runtime.profiling.Percent', 'relax.ExecutableLoadFromFile', 'runtime.ModuleLoadStaticLibrary', 'runtime.SystemLib', 'runtime.config_threadpool', 'runtime.NumThreads', 'runtime.GetNumOfGlobals', 'runtime.GetGlobalFields', 'runtime.GetNumOfPrimitives', 'runtime.GetPrimitiveFields', 'runtime.Load_Executable', 'runtime._VirtualMachine', 'device_api.rpc', 'rpc.CreateEventDrivenServer', 'rpc.ServerLoop', 'rpc.LocalSession', 'runtime.RPCTimeEvaluator', 'cache_flush_cpu_non_first_arg', 'tvm.rpc.server.ImportModule', 'tvm.rpc.server.ModuleGetFunction', 'rpc.LoadRemoteModule', 'rpc.ImportRemoteModule', 'rpc.SessTableIndex', 'tvm.rpc.NDArrayFromRemoteOpaqueHandle', 'tvm.rpc.server.upload', 'tvm.rpc.server.download', 'rpc.Connect', 'tvm.graph_executor.create', 'tvm.graph_executor_factory.create', 'runtime.module.loadbinary_GraphExecutorFactory', 'runtime.module.loadbinary_GraphRuntimeFactory', 'tvm.graph_executor_debug.create', 'runtime._VirtualMachineDebug', 'tvm.aot_executor_factory.create', 'runtime.module.loadbinary_AotExecutorFactory', 'tvm.contrib.random.randint', 'tvm.contrib.random.normal', 'tvm.contrib.random.random_fill', 'tvm.contrib.sort.argsort_nms', 'tvm.contrib.sort.argsort', 'tvm.contrib.sort.sort', 'support.GetLibInfo', 'diagnostics.override_renderer', 'tvm.default_trace_action', 'target._load_config_dict', 'tvm_callback_rocm_link', 'tvm_callback_rocm_bitcode_path', 'tvm_callback_cuda_compile', 'tvm_callback_libdevice_path', 'tvm_callback_sdaccel_compile']

**Reply by junrushao** [reply]
这个脚本里面显示出来的问题是，似乎我们 windows 默认载入了 libtvm_runtime 而不是完整的 compiler，导致 compiler 部分的 global function 找不到。CC: @tqchen

**Comment by Relph1119** [comment]
Window Python 3.10.5
同样也发现了这个问题
